### PR TITLE
inital script commit

### PR DESCRIPTION
**Pull request by AmitBEL**

### CODE REVIEW

File: change_all_equations_to_text.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+def zip_to_docx(zip_file_path):
+    docx_file_path = zip_file_path[0:-4] + "_f.docx"
+    os.rename(zip_file_path, docx_file_path)
+    # copyfile(docx_file_path, zip_file_path)
+    return docx_file_path
+
+
+
+# print(sys.argv[1])
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: please remove all of the unessery comments.

File: change_all_equations_to_text.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+equations_to_text(document_xml_file_path1, tmp_document_xml_file_path1)
+os.replace(tmp_document_xml_file_path1, document_xml_file_path1)
+zip_file(zip_file_path1, extract_directory1)
+new_docx_file_path1 = zip_to_docx(zip_file_path1)
+shutil.rmtree(extract_directory1)
+
+
+# xml_file_path1 = "C:/Users/mitmi/Desktop/Equations Project/Test/equations/word/document.xml"
+# new_xml_file_path1 = "C:/Users/mitmi/Desktop/Equations Project/Test/equations/word/document1.xml"
+# equations_to_text(xml_file_path1, new_xml_file_path1)
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: please remove.

File: change_all_equations_to_text.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    docx_file_path = zip_file_path[0:-4] + "_f.docx"
+    os.rename(zip_file_path, docx_file_path)
+    # copyfile(docx_file_path, zip_file_path)
+    return docx_file_path
+
+
+
+# print(sys.argv[1])
+orig_docx_file_path1 = sys.argv[1]
+docx_file_path1 = os.path.normpath(orig_docx_file_path1)
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: r'asdasdas/asdasd/asdasd'

File: change_all_equations_to_text.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+def zip_to_docx(zip_file_path):
+    docx_file_path = zip_file_path[0:-4] + "_f.docx"
+    os.rename(zip_file_path, docx_file_path)
+    # copyfile(docx_file_path, zip_file_path)
+    return docx_file_path
+
+
+
+# print(sys.argv[1])
+orig_docx_file_path1 = sys.argv[1]
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: change to original

Comment (Moshikolev) [reply change_all_equations_to_text.py]: ORIGINAL_DOCX_PATH_INDEX = 1

File: change_all_equations_to_text.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                p = Path(file_path)
+                zipObj.write(file_path, p.relative_to(directory))
+
+
+def zip_to_docx(zip_file_path):
+    docx_file_path = zip_file_path[0:-4] + "_f.docx"
+    os.rename(zip_file_path, docx_file_path)
+    # copyfile(docx_file_path, zip_file_path)
+    return docx_file_path
+
```
Comment (Moshikolev) [inline change_all_equations_to_text.py:73]: Add comments here about the logic of the function
1. getting the docx file
2. extracting to zop
3. etc ...

File: change_all_equations_to_text.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+def equations_to_text(document_xml_file_path, out_xml_file_path):
+    with open(document_xml_file_path, "rt", encoding="utf8") as fin:
+        with open(out_xml_file_path, "wt", encoding="utf8") as fout:
+            for line in fin:
+                fout.write(line.replace('<m:r><w:rPr>', '<m:r><m:rPr><m:nor /></m:rPr><w:rPr>'))
+
+
+def zip_file(zip_file_path, directory):
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: please add to all functions parameters an '_'

File: change_all_equations_to_text.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+
+# print(sys.argv[1])
+orig_docx_file_path1 = sys.argv[1]
+docx_file_path1 = os.path.normpath(orig_docx_file_path1)
+# docx_file_path1 = "C:/Users/mitmi/Desktop/Equations Project/Test/equations.docx"
+zip_file_path1 = docx_to_zip(docx_file_path1)
+extract_directory1 = extract_zip_file(zip_file_path1)
+document_xml_file_path1 = extract_directory1 + 'word/document.xml'
+tmp_document_xml_file_path1 = extract_directory1 + 'word/document1.xml'
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: remove the file from the names

File: change_all_equations_to_text.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+equations_to_text(document_xml_file_path1, tmp_document_xml_file_path1)
+os.replace(tmp_document_xml_file_path1, document_xml_file_path1)
+zip_file(zip_file_path1, extract_directory1)
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: extract_directory => please think on a better name

File: change_all_equations_to_text.py [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+new_docx_file_path1 = zip_to_docx(zip_file_path1)
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: new_docx_file_path1 => result_docx_path

File: change_all_equations_to_text.py [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+shutil.rmtree(extract_directory1)
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: please add a comment explaining what this line does

File: change_all_equations_to_text.py [hunk 11]
```diff
@@ -0,0 +1,64 @@
+import sys
+import os
+import zipfile
+import shutil
+from pathlib import Path
+
+
+def docx_to_zip(docx_file_path):
+    zip_file_path = docx_file_path[0:-4] + "zip"
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: please add comments on what the function does 
"""

File: change_all_equations_to_text.py [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+    new_docx_file_path = docx_file_path + '_copy'
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: please add to the comment that you are creating a new file to preserve the original one

Comment (Moshikolev) [reply change_all_equations_to_text.py]: new_docx_file_path => duplicated_docx_path

File: change_all_equations_to_text.py (hunk 11 above)
Comment (Moshikolev) [inline change_all_equations_to_text.py]: please try to copy the file directly to the zip path

File: change_all_equations_to_text.py [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    shutil.copyfile(docx_file_path, new_docx_file_path)
+    os.rename(new_docx_file_path, zip_file_path)
+    return zip_file_path
+
+
+def extract_zip_file(zip_file_path):
+    with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
+        extract_directory = zip_file_path[0:-4] + '/'
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: extracted_directory_path

File: change_all_equations_to_text.py [hunk 14]
```diff
@@ -0,0 +1,64 @@
+import sys
+import os
+import zipfile
+import shutil
+from pathlib import Path
+
+
```
Comment (Moshikolev) [inline change_all_equations_to_text.py:7]: replace_file_suffix(suffix_length,new_suffix)
replace_file_suffix(suffix_length=4,new_suffix="zip")

File: change_all_equations_to_text.py [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+def extract_zip_file(zip_file_path):
+    with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
+        extract_directory = zip_file_path[0:-4] + '/'
+        zip_ref.extractall(extract_directory)
+        return extract_directory
+
+
+def equations_to_text(document_xml_file_path, out_xml_file_path):
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: document_xml_path

File: change_all_equations_to_text.py [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+    with open(document_xml_file_path, "rt", encoding="utf8") as fin:
+        with open(out_xml_file_path, "wt", encoding="utf8") as fout:
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: please check what is this t

File: change_all_equations_to_text.py (hunk 6 above)
Comment (Moshikolev) [inline change_all_equations_to_text.py]: create_zip_file

File: change_all_equations_to_text.py [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+                fout.write(line.replace('<m:r><w:rPr>', '<m:r><m:rPr><m:nor /></m:rPr><w:rPr>'))
+
+
+def zip_file(zip_file_path, directory):
+    with zipfile.ZipFile(zip_file_path, 'w') as zipObj:
+        for folder_name, sub_folders, filenames in os.walk(directory):
+            for filename in filenames:
+                file_path = os.path.join(folder_name, filename)
+                p = Path(file_path)
+                zipObj.write(file_path, p.relative_to(directory))
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: please add comment about the relative

File: change_all_equations_to_text.py [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+
+
+def zip_to_docx(zip_file_path):
+    docx_file_path = zip_file_path[0:-4] + "_f.docx"
+    os.rename(zip_file_path, docx_file_path)
+    # copyfile(docx_file_path, zip_file_path)
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: please remove

**Review by Moshikolev — CHANGES_REQUESTED** [review]

File: change_all_equations_to_text.py (hunk 16 above)
Comment (AmitBEL) [reply change_all_equations_to_text.py]: Open in text mode.

File: change_all_equations_to_text.py [hunk 19]
```diff
@@ -0,0 +1,119 @@
...
+from pathlib import Path
+
+
+def replace_path_suffix(path, _suffix_length, _new_suffix):
+    """
+    Replace the suffix of a path with new suffix
+    :param path:            original path
+    :param _suffix_length:  the number of chars to be deleted
+                            from the end of path
+    :param new_suffix:      the new suffix to be added to path
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: add underscore

File: change_all_equations_to_text.py [hunk 20]
```diff
@@ -0,0 +1,119 @@
...
+
+def extract_zip_file(_zip_path):
+    """
+    Extract zip file to a directory with the same name as the zip file
+    :param _zip_path: the zip file path to be extracted
+    :return:          the extraction directory path
+    """
+    with zipfile.ZipFile(_zip_path, 'r') as zip_ref:
+        _extraction_directory = replace_path_suffix(_zip_path, _suffix_length=4, _new_suffix="/")
+        # _extraction_directory = _zip_path[0:-4] + '/'
```
Comment (Moshikolev) [inline change_all_equations_to_text.py]: please remove.

**Review by Moshikolev — CHANGES_REQUESTED** [review]

File: change_all_equations_to_text.py (hunk 20 above)
Comment (Moshikolev) [reply change_all_equations_to_text.py]: Please add a few words in the readme file

**Review by Moshikolev — APPROVED** [review]

**Pull request merged**